### PR TITLE
chore: clean up HCP Vault Secrets OpenAPI work post-launch

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Cleans up config flags for HCP Vault Secrets, and executes cleanup and configuration organization for OpenApiDocsView as well, as the latter ties in closely with flag-related cleanup.

## 🤷 Why

To clean up now-irrelevant configuration, and detangle the new `OpenApiDocsView` from the previous `ApiDocsView`. Intent here is also to solidify the `OpenApiDocsView` setup as we move into onboarding additional products to using the view.

## 🛠️ How

- Lifts up API docs related utilities to `lib/api-docs`, updates import paths accordingly
- Refactors the HCP Vault Secrets page file to remove flags, and collect "page configuration" in one place
- Cleans up now-unused HCP Vault Secrets flag

## 🧪 Testing

Each of the below pages should look and function as they do upstream:

- [ ] [/hcp/api-docs/vault-secrets] (compare to [upstream][upstream/hcp/api-docs/vault-secrets])
- [ ] [/boundary/api-docs] (compare to [upstream][upstream/boundary/api-docs])
- [ ] [/waypoint/api-docs] (compare to [upstream][upstream/waypoint/api-docs])
- [ ] [/hcp/api-docs/packer] (compare to [upstream][upstream/hcp/api-docs/packer])

## 💭 Anything else?

Not at the moment!

[task]: https://app.asana.com/0/1204678746647847/1205302765084538/f
[preview]: https://dev-portal-git-zshcp-vault-openapi-cleanup-hashicorp.vercel.app/
[/hcp/api-docs/vault-secrets]: https://dev-portal-git-zshcp-vault-openapi-cleanup-hashicorp.vercel.app/hcp/api-docs/vault-secrets
[/boundary/api-docs]: https://dev-portal-git-zshcp-vault-openapi-cleanup-hashicorp.vercel.app/boundary/api-docs
[/waypoint/api-docs]: https://dev-portal-git-zshcp-vault-openapi-cleanup-hashicorp.vercel.app/waypoint/api-docs
[/hcp/api-docs/packer]: https://dev-portal-git-zshcp-vault-openapi-cleanup-hashicorp.vercel.app/hcp/api-docs/packer
[upstream/hcp/api-docs/vault-secrets]: https://developer.hashicorp.com/hcp/api-docs/vault-secrets
[upstream/boundary/api-docs]: https://developer.hashicorp.com/boundary/api-docs
[upstream/waypoint/api-docs]: https://developer.hashicorp.com/waypoint/api-docs
[upstream/hcp/api-docs/packer]: https://developer.hashicorp.com/hcp/api-docs/packer

### CODE REVIEW

File: config/base.json [hunk 1]
```diff
@@ -37,7 +37,6 @@
 	},
 	"flags": {
 		"enable_datadog": true,
-		"enable_io_beta_cta": true,
-		"enable_hcp_vault_secrets_api_docs_revision": false
```
Comment (zchsh) [inline config/base.json:41]: The `enable_hcp_vault_secrets_api_docs_revision` flag is no longer used, so I've removed it completely 👍

File: src/lib/api-docs/index.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+export { fetchCloudApiVersionData } from './fetch-cloud-api-version-data'
```
Comment (zchsh) [inline src/lib/api-docs/index.ts:6]: I've moved all the `api-docs` utils I was previously "cross importing" from the "old view" to the "new view" up to `lib`. 

Detangling the two views is meant to add clarity to what the new view consumes, and should also make it easier to fully deprecate and remove the old view code once we're ready.

File: src/pages/hcp/api-docs/vault-secrets/[[...page]].tsx [hunk 3]
```diff
@@ -3,150 +3,78 @@
...
-		href: '/tutorials/library?product=vault&edition=hcp',
-	},
-	{
-		title: 'Certifications',
-		href: '/certifications/security-automation',
-	},
-	{
-		title: 'Community',
-		href: 'https://discuss.hashicorp.com/',
+const PAGE_CONFIG: OpenApiDocsPageConfig = {
```
Comment (zchsh) [inline src/pages/hcp/api-docs/vault-secrets/[[...page]].tsx:26]: I've refactored this page file to remove the conditional previously controlled by the `enable_hcp_vault_secrets_api_docs_revision` flag.

I've also collected all page-specific configuration in this `PAGE_CONFIG` variable. Intent is to make it easier to copy-pasta this page file to a new location, and be able to modify all the config needed to stand up a new product's OpenApiDocsView in one place.

File: src/views/open-api-docs-view/types.ts [hunk 4]
```diff
@@ -163,3 +164,39 @@ export interface OpenApiDocsViewProps {
 	 */
 	_placeholder: $TSFixMe
 }
+
+/**
+ * Type definition for OpenApiDocsView server-side page configuration
+ */
+export interface OpenApiDocsPageConfig {
```
Comment (zchsh) [inline src/views/open-api-docs-view/types.ts:171]: I've added a type definition for `OpenApiDocsPageConfig` - it felt useful to clarify what each part of the page configuration does (previously these were comments inline in the page file).

File: config/base.json (hunk 1 above)
Comment (kendallstrautman) [reply config/base.json:41]: Thanks for cleaning this up

**Review by kendallstrautman — APPROVED** [review]
Tested everything, LGTM! 🧼

**Pull request merged**